java/arcs/core/analysis/RecipeGraph.kt

```diff
@@ -53,6 +69,27 @@ class RecipeGraph(recipe: Recipe) {
         return particleNode
     }
 
+    /**
+     * A class representing the properties of a join.
+     *
+     * This is a dummy class for the time being, but will have more information when the support
+     * for join is fully flshed out. This should also move out of this file. */
+    data class JoinSpec(val type: String = "INNER")
```

Comment:
It's not entirely clear to me how this will work TBH. In the immediate future we will be working on inner joins that have a refinement described on the tuple instances inside (e.g. `first.someId == second.someId`) but if my proposal for expressions will move forward then I would argue against reflecting outer joins in this way, as the expression for joining with an outer join is not a predicate on an instance, so it breaks out abstractions IMO.

How would we reflect exactly the outer joins with these expressions that I'm proposing is a bit TBD, definitely tractable, but I haven't gone that far in design yet.

Reply:
Thanks for taking a look, Piotr! This is mainly a place holder for now. As the design of join evolves,  data flow analysis will need additional information, which I expect something like `JoinSpec` will provide. Just wanted to bring this to your attention.